Clarify new, open, and edit command semantics

hschne opened the pull request:
### Metadata

This PR fixes #817 and fixes #420.

### Problem / Motivation

`open` and `edit` were only aliases of `new`, which made them harder to
discover in help and blurred their intended roles.

- `edit` is an explicit command for opening an existing project file
- `open` is an explicit command for the existing create-or-open editor flow
- `new` keeps its prior behavior, including `new PROJECT SESSION`

README and changelog text were updated to match the command behavior.

### Testing

Verify `edit` opens an existing config and fails for a missing one:

```sh
tmuxinator edit project1
```

Verify `open` creates a config if missing, or opens it if it already exists:

```sh
tmuxinator open project1
```

Verify `new` still behaves as before:

```sh
tmuxinator new project1
tmuxinator new imported-project existing-session
```

Confirm the command help output:

```sh
tmuxinator help open
tmuxinator help edit
```

hschne committed:
Split out edit command

hschne committed:
Update changelog and cli

hschne commented:
@akofink It's been a while since I worked on this. As far as I can tell all good, thanks for picking this up :raised_hands:

akofink committed:
revert: restore pre-split open and new behavior

Metadata

- Branch: 25-04-split-out-edit-and-open-commands
- Base preserved: author commits 829e686 and f4c98b2
- Scope: command surface and changelog only

Problem

- The author branch changes both the command split and the user-facing semantics for new, open, and edit, which makes the follow-up fix set much harder to review.
- To reduce scope, the next commit should show only the minimal change needed for the historical edit issues instead of carrying the broader command redesign.

Solution

- Restore lib/tmuxinator/cli.rb and CHANGELOG.md to the pre-branch behavior from upstream/master.
- Keep the author commits in history, but clear their command-surface effect so the next commit can reintroduce only the focused edit fix.

akofink committed:
fix: reintroduce explicit edit command

Metadata

- Branch: 25-04-split-out-edit-and-open-commands
- Base preserved: author commits 829e686 and f4c98b2
- Scope: explicit edit command, focused tests, and small docs updates

Problem

- The historical issues behind this branch are narrower than the full new/open/edit redesign.
- Users need an edit command that appears in help and opens an existing config without generating or clobbering a project file.
- Keeping open and new semantics unchanged reduces review risk.

Solution

- Reintroduce edit as an explicit CLI command with the e alias while leaving new and open behavior unchanged.
- Make edit open an existing named or local config and exit with an error when the target does not exist.
- Add focused CLI coverage for edit help and missing-file behavior, and update the README and changelog to match the smaller command surface.

akofink committed:
fix: reintroduce explicit open command

Metadata

- Branch: 25-04-split-out-edit-and-open-commands
- Base preserved: author commits 829e686 and f4c98b2
- Scope: explicit open command, focused tests, and small docs updates

Problem

- Open is still only an alias of new, so it does not have its own help entry or an explicit command definition.
- To keep scope small and reviewable, open should become explicit without changing the underlying create-or-open behavior it already had.

Solution

- Reintroduce open as an explicit CLI command with the o alias while keeping its existing create-or-open behavior by delegating to new_project.
- Add focused CLI coverage for open help and explicit open behavior for both missing and existing project files.
- Update the README and changelog so they describe open as an explicit command instead of an alias.

akofink commented on lib/tmuxinator/cli.rb:
```diff
@@ -101,6 +97,49 @@ def new(name = nil, session = nil)
+        return
+      end
+
+      path = config_path(name, local: options[:local])
+      if File.exist?(path)
+        Kernel.system("$EDITOR #{path}") || doctor
+      elsif options[:local] && name.nil?
+        exit! Tmuxinator::Config::NO_LOCAL_FILE_MSG
+      else
+        exit! "Project #{name} doesn't exist!"
+      end
```
Scope creep; This behavior is unchanged from master. It should be fixed but not with this PR IMO.

akofink commented on lib/tmuxinator/cli.rb:
```diff
@@ -101,6 +97,49 @@ def new(name = nil, session = nil)
+      end
+
+      path = config_path(name, local: options[:local])
+      if File.exist?(path)
+        Kernel.system("$EDITOR #{path}") || doctor
+      elsif options[:local] && name.nil?
```
Scope creep; This behavior is unchanged from master. It should be fixed but not with this PR IMO.

akofink committed:
fix: clarify local option help text

Metadata

- Branch: 25-04-split-out-edit-and-open-commands
- Scope: CLI help text and focused help-output specs

Problem

- The --local option description said it would create a local project file even for commands like edit that do not create files.
- That wording overstated the behavior and made the help output misleading.

Solution

- Change the shared --local wording to the neutral phrase 'Use local project file at ./.tmuxinator.yml'.
- Add focused help-output assertions for new, open, and edit so the wording stays aligned with actual behavior.
